src/ia-bookmarks-list.js

```diff
@@ -0,0 +1,86 @@
+import { nothing } from 'lit-html';
+import { repeat } from 'lit-html/directives/repeat.js';
+import { classMap } from 'lit-html/directives/class-map.js';
+import { html, LitElement } from 'lit-element';
+import '@internetarchive/icon-edit-pencil/icon-edit-pencil';
+import bookmarksListCSS from './styles/ia-bookmarks-list.js';
+
+export class IABookmarksList extends LitElement {
+  static get styles() {
+    return bookmarksListCSS;
+  }
+
+  static get properties() {
+    return {
+      activeBookmarkID: { type: Number },
+      bookmarks: { type: Array },
+      renderHeader: { type: Boolean },
+    };
+  }
+
+  constructor() {
+    super();
+    this.activeBookmarkID = undefined;
+    this.bookmarks = [];
+    this.renderHeader = false;
+  }
+
+  emitEditEvent(e, bookmark) {
+    e.stopPropagation();
+    this.dispatchEvent(new CustomEvent('bookmarkEdited', {
+      bubbles: true,
```

Comment:
if we are catching these events targeting its node, then we do not need to bubble further up

Reply:
That's true. I was thinking BookReader might have a listener on the document for updating the state of a given page's bookmark button, but that could be handled by the controller instead.